src/Symfony/Component/Runtime/autoload.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Symfony\Component\Runtime\SymfonyRuntime;
+
+if (true === require_once dirname(__DIR__, 2).'/autoload.php') {
+    return;
+}
+
+if (!isset($_SERVER['APP_RUNTIME'])) {
+    $runtime = new SymfonyRuntime();
+} elseif (is_string($runtime = $_SERVER['APP_RUNTIME'])) {
+    $runtime = new $runtime();
+}
+
+if (1 === $main = require $_SERVER['SCRIPT_FILENAME']) {
+    exit(0);
+}
+
+[$main, $arguments] = $runtime->resolveMain($main);
+$result = $main(...$arguments);
+$handler = $runtime->getHandler($result);
+
+exit($handler($result));
```

Comment:
At first, I didn't like the naming here at all. `$main` is used twice for potentially different things. One could do `s/main/kernel/` but that only make sense for a Symfony application...

I dont know any better names because it is too flexible =)


```php
// $result could be a Response, a Kernel or pretty much anything.
$result = $main(...$arguments);
```

Reply:
naming things... :)
"main" is the term used for this concept in C, Rust, etc. Before/after resolving, it's still the same concept. No need for another name.
About the rest, that's the best I could come up with and they're good to me (I can tell that's not my first iteration at naming these...)

Reply:
I believe that you iterated the names a few times. I tried myself without any success to come up with something better.